remoting/src/main/java/org/apache/rocketmq/remoting/util/ServiceProvider.java

```diff
@@ -109,15 +114,11 @@ protected static InputStream getResourceAsStream(ClassLoader loader, String name
                 }
                 String serviceName = reader.readLine();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *reader*. The resource is referenced in statements at the following lines: 108, 111, 113, 115, 122. The resource closure statement is at line: 124. There are other execution paths that do not contain closure statements, e.g., when exception is thrown by BufferedReader.readLine. Either a) close *reader* in a try-finally block or b) close the resource by declaring *reader* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).